src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -1454,6 +1454,63 @@ Usage example:
   }
 
 
+Cache for computations
+~~~~~~~~~~~~~~~~~~~~~~
+
+The dashboard backend provides a cache infrastructure to store values of
+computations that might take a long time to process.
+
+We provide the ``Cache`` class to help developers interacting with the
+cache infrastructure in the following way::
+
+  from ..tools import Cache
+  # ...
+  Cache.register(cache_id, func, interval)
+  # ...
+  Cache.get(cache_id, timeout)
+
+Method ``register`` registers a new cache entry with the computation
+implemented by ``func`` and identified by ``cache_id``.
+
+* ``cache_id`` is a string that will be used as the key to register the
+  computation in the cache.
+
+* ``func`` is a python function that computes and returns values that are
+  stored in the cache.
+
+* ``interval`` the number of seconds between each execution of ``func``.
+  When seting this argument the cache will automatically refresh the cache
+  values by running the computation periodically.
+  If this argument is set to ``None`` (default value), the cache values are
+  only refreshed when consulting the cache values with the `Cache.get``
+  method.
+
+Method ``get`` returns the values present on the cache for the key
+``cache_id``.
+This method has different behaviors dependending if the cache entry is
```

Comment:
typo: dependening -> depending
proposal: ..depending on whether the cache entry is..